c_src/workitems.h

```diff
@@ -629,6 +619,8 @@ namespace eleveldb {
         std::string end_key_;
         bool has_end_key_;
         SyncObject * sync_obj_;
```

Comment:
Appears sync_obj_ is NOT optional.  Therefore it needs to be a reference, not a pointer.  If optional, pointer is not validated for NULL and could cause segfault.